libs/dal/src/repositories/job/job.repository.ts

```diff
@@ -174,27 +175,42 @@ export class JobRepository extends BaseRepository<JobDBModel, JobEntity, Enforce
   public async shouldDelayDigestJobOrMerge(
     job: JobEntity,
     digestKey?: string,
-    digestValue?: string | number
-  ): Promise<{ matched: number; modified: number }> {
-    const execution = {
-      matched: 0,
-      modified: 0,
-    };
+    digestValue?: string | number,
+    digestMeta?: IDigestRegularMetadata
+  ): Promise<DigestCreationResultEnum> {
+    const isBackoff = job.digest?.type === DigestTypeEnum.BACKOFF || (job.digest as IDigestRegularMetadata)?.backoff;
```

Comment:
We are keeping the fallback of the shortcut to handle the existing jobs in Redis when deploying this change? If so, well thought. 👏🏻